Tests/SwiftFormatTests/Core/IgnoreFileTests.swift

```diff
@@ -0,0 +1,41 @@
+import SwiftFormat
+import XCTest
+
+final class IgnoreFileTests: XCTestCase {
+
+  func testMissingIgnoreFile() throws {
+    let url = Bundle.module.url(forResource: "missing", withExtension: "", subdirectory: "Ignore Files")
+    XCTAssertNotNil(url)
+    XCTAssertNil(try IgnoreFile(forDirectory: url!))
+    XCTAssertNil(try IgnoreFile(for: url!.appending(path:"file.swift")))
+  }
+
+  func testValidIgnoreFile() throws {
+    let url = Bundle.module.url(forResource: "valid", withExtension: "", subdirectory: "Ignore Files")
+    XCTAssertNotNil(url)
+    XCTAssertNotNil(try IgnoreFile(forDirectory: url!))
+    XCTAssertNotNil(try IgnoreFile(for: url!.appending(path:"file.swift")))
+  }
+
+  func testInvalidIgnoreFile() throws {
+    let url = Bundle.module.url(forResource: "invalid", withExtension: "", subdirectory: "Ignore Files")
+    XCTAssertNotNil(url)
+    XCTAssertThrowsError(try IgnoreFile(forDirectory: url!))
+    XCTAssertThrowsError(try IgnoreFile(for: url!.appending(path:"file.swift")))
+  }
+
+  func testEmptyIgnoreFile() throws {
+    let url = Bundle.module.url(forResource: "empty", withExtension: "", subdirectory: "Ignore Files")
```

Comment:
I am personally not a fan of test resources, especially if they are pretty small like for these test cases because it means that you can’t reason about the test case by just looking at the test case anymore but need to inspect the resource files. Could you change the tests to write these ignore files to a temporary directory from a string literal instead?

Edit: If we have an initializer of `IgnoreFile` that takes a string, we might not  need to write the ignore files to the file system in some of these tests.